diff --git a/mlx/backend/metal/matmul.cpp b/mlx/backend/metal/matmul.cpp
index 98d7b668..b7a1d128 100644
--- a/mlx/backend/metal/matmul.cpp
+++ b/mlx/backend/metal/matmul.cpp
@@ -352,7 +352,7 @@ void steel_matmul_regular_axpby(
     float alpha /* = 1.0f */,
     float beta /* = 0.0f */) {
   if (metal::is_nax_available() && !issubdtype(a.dtype(), complexfloating) &&
-      (env::enable_tf32() || a.dtype() != float32)) {
+      (env::enable_tf32() && a.dtype() == float32)) {
     return steel_matmul_regular_axpby_nax<CHECK_AB>(
         /* const Stream& s = */ s,
         /* metal::Device& d = */ d,
@@ -949,7 +949,7 @@ void steel_matmul_axpby(
   constexpr int min_k_threshold = 10240;
   if (batch_size_out == 1 && metal::is_nax_available() &&
       !issubdtype(a.dtype(), complexfloating) &&
-      (env::enable_tf32() || a.dtype() != float32) &&
+      (env::enable_tf32() && a.dtype() == float32) &&
       int64_t(M) * N >= min_mn_threshold && K >= min_k_threshold &&
       K >= (3 * std::max(M, N))) {
     return steel_gemm_splitk_axpby_nax<CHECK_AB>(
@@ -2388,7 +2388,7 @@ void GatherMM::eval_gpu(const std::vector<array>& inputs, array& out) {
   // matmuls and reuse reading a and b.
   if (M == 1 && right_sorted_ == true) {
     if (metal::is_nax_available() &&
-        (env::enable_tf32() || a.dtype() != float32)) {
+        (env::enable_tf32() && a.dtype() == float32)) {
       return gather_mm_rhs_nax(a, b, rhs_indices, out, d, s);
     }
     gather_mm_rhs(a, b, rhs_indices, out, d, s);
